interp/src/primitives/combinational.rs

```diff
@@ -55,10 +58,53 @@ impl Primitive for StdConst {
 }
 
 // ===================== New core ======================
-comb_primitive!(StdMux[WIDTH](cond: WIDTH, left: WIDTH, right: WIDTH) -> (out: WIDTH) {
-    todo!()
-    //cond isn't width wide, it's 1... but that gives compiler error, why?
-});
+
+pub struct StdMux {
```

Comment:
Is there a reason this can't be implemented using the macro that defines other combinational primitives?

Reply:
The macro can't accept constants right now and the cond input is 1 bit wide for this primitive, not WIDTH, so if written with the macro it will break whenever we try to use it. I can probably hack the macro to allow constants in some positions, but I figured it would be faster to just implement it by hand.